src/main/java/com/pmi/SchemaCompiler/SchemaGeneratorMojo.java

```diff
@@ -0,0 +1,103 @@
+package com.pmi.SchemaCompiler;
+
+import com.pmi.SchemaCompiler.data.Enum;
+import com.pmi.SchemaCompiler.data.Schema;
+import com.pmi.SchemaCompiler.data.Type;
+import com.pmi.SchemaCompiler.utils.YamlUtil;
+import com.samskivert.mustache.Mustache;
+import com.samskivert.mustache.Template;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.List;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+
+/** Goal which generate source files of data modle based on the definiation in schema yaml files. */
+@Mojo(name = "generate-schema", defaultPhase = LifecyclePhase.PROCESS_SOURCES)
+public class SchemaGeneratorMojo extends AbstractMojo {
+  // The resource directory which should contain a schema folder having main.yml inside it.
+  @Parameter(defaultValue = "${project.build.resources[0].directory}", required = true)
+  private File resourceDir;
+
+  @Parameter(defaultValue = "${project.build.sourceDirectory}", required = true)
+  private File sourceDir;
+
+  public void execute() throws MojoExecutionException {
+    try {
+      Path schemaDirPath = Paths.get(resourceDir.getPath(), "schema");
```

Comment:
this is highly coupled, can we pass a parameter?

Reply:
Yes, we can mark `schemaDirPath` as a parameter.